substrate/bin/node/runtime/src/lib.rs

```diff
@@ -2560,11 +2572,37 @@ pub type SignedExtra = (
 		pallet_asset_conversion_tx_payment::ChargeAssetTxPayment<Runtime>,
 	>,
 	frame_metadata_hash_extension::CheckMetadataHash<Runtime>,
+	pallet_revive::evm::runtime::CheckEthTransact<Runtime>,
 );
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct EthSignedExtra;
+
+impl EthExtra for EthSignedExtra {
+	type Config = Runtime;
+	type Extra = SignedExtra;
+
+	fn get_eth_transact_extra(nonce: u32, eth_fee: u128) -> Self::Extra {
+		(
+			frame_system::CheckNonZeroSender::<Runtime>::new(),
+			frame_system::CheckSpecVersion::<Runtime>::new(),
+			frame_system::CheckTxVersion::<Runtime>::new(),
+			frame_system::CheckGenesis::<Runtime>::new(),
+			frame_system::CheckEra::from(crate::generic::Era::Immortal),
+			frame_system::CheckNonce::<Runtime>::from(nonce),
+			frame_system::CheckWeight::<Runtime>::new(),
+			pallet_asset_conversion_tx_payment::ChargeAssetTxPayment::<Runtime>::from(0, None)
+				.into(),
+			frame_metadata_hash_extension::CheckMetadataHash::<Runtime>::new(false),
+			pallet_revive::evm::runtime::CheckEthTransact::<Runtime>::from(eth_fee),
+		)
+	}
+}
+
 /// Unchecked extrinsic type as expected by this runtime.
 pub type UncheckedExtrinsic =
-	generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
+	pallet_revive::evm::runtime::UncheckedExtrinsic<RuntimeCall, EthSignedExtra>;
```

Comment:
I don't get this. Shouldn't this break all wallets since you replace `SignedExtra` (type that contains data) with `EthSignedExtra` (zero sized type) for all extrinsics?

Reply:
It actually uses `EthSignedExtra::Extra` and hence leaves the on wrote format of the extrinsic untouched.